Add github-review

charignon opened the pull request:
### Brief summary of what the package does
`greview` lets you submit Github code review with Emacs.

With `greview-start-review` you can pull the content of a pull request into a buffer as a diff with comments corresponding to the PR description. In that buffer you can add comment (global and inline) that you want to make on the pull request. Once done, you can submit these comments as a code review with one of `greview-approve`, `greview-comment` and `greview-reject`.

### Direct link to the package repository
https://github.com/charignon/greview

### Your association with the package
I am the maintainer of greview

### Relevant communications with the upstream package maintainer
None needed

### Checklist

Please confirm with `x`:

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [X] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] `M-x checkdoc` is happy with my docstrings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

charignon commented:
Thanks for the review :), I fixed the issues that you raised:
- https://github.com/charignon/greview/commit/1acd0f6 Add spacing in interactive message
- https://github.com/charignon/greview/commit/9f1f950 Fix indentation mismatch
- https://github.com/charignon/greview/commit/593a274 Capitalize GitHub properly
- https://github.com/charignon/greview/commit/231fd95 Update commentary

The error you ran into is probably because your GHE token isn't set up properly. Could you confirm that you followed the steps under **Configuration** at https://github.com/charignon/greview?

riscy commented on recipes/greview:
```diff
@@ -0,0 +1 @@
+(greview :repo "charignon/greview" :fetcher github)
```
Experimental inline comment using `greview`.

riscy commented:
Thanks for accommodating my fumbling.  This is really nice.

I looked into the `ghub` authentication docs and ran the command `ghub-create-token`.   When it asks me to define the package, I put `greview` and get the error: `call-interactively: Symbol’s value as variable is void: greview-github-token-scopes` -- so I was probably wrong to do that.

If I set the package to `ghub`, I get a line like `machine api.github.com login riscy^ghub ...` in my `authoinfo` which seems to work, but doesn't match the documentation in your repo.

charignon commented:
I created the token manually through the GitHub UI and added it to my authinfo file. I checked the procedure on a new host and it worked. I updated the instructions accordingly (https://github.com/charignon/greview/commit/39e87569bd2cb65d9ee9dd9d9267e47f50e4ce8f), thanks!

tarsius commented:
From `ghub`'s manual:

> Every package should use its own token.  This allows you as the author
> of some package to only request access to API scopes that are actually
> needed, which in turn might make it easier for users to trust your
> package not to do unwanted things.
> 
>    The scopes used by ‘PACKAGE’ have to be defined using the variable
> ‘PACKAGE-github-token-scopes’, and you have to tell ‘ghub-request’ on
> behalf of which package a request is being made by passing the symbol
> ‘PACKAGE’ as the value of its ‘AUTH’ argument.
> 
>      (ghub-request "GET" "/user" nil :auth 'PACKAGE)

See the rest of the "Using Ghub in a Package" node for more information.

charignon commented:
Thanks for the pointer @tarsius and sorry for missing this in the first place. I added and tested the scope definition https://github.com/charignon/greview/commit/6b54cde67f52c8488b12ad974895cb8e6ded1144.

charignon commented:
@riscy could you please take another look when you get a chance?

mattiasb commented:
One small suggestion: The "g" prefix doesn't say much. Would it make sense to call the package github-review? Or (of there are plans to support more forges): git-forge-review?

charignon commented:
@mattiasb thanks for the feedback, you are right the `g` prefix does not say much. I don't intend on adding support to more forges for now, renaming it to `github-review` is a great idea! I will work on that.

charignon commented:
@riscy I renamed the package to `github-review`, this is ready for another round of review :) Thank you all for your work on this.

charignon commented:
Sure, that's a good idea for later improvements. I applied your suggestion (see https://github.com/charignon/github-review/commit/f2982aa91694c3b9cfd67fdb077f999cb9970c22). Thanks for all the review, what is missing to get this merged?

riscy commented:
I'm just giving it a first pass, I'm not a MELPA maintainer.  :)

purcell commented:
Thanks everyone for the review assistance.

Nice work @charignon - this is pretty epic! Looks good to me so merging now. Welcome to MELPA! :tada:
